src/lp_sensitivity2.jl

```diff
@@ -0,0 +1,429 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SparseArrays
+import LinearAlgebra
+
+"""
+    SensitivityReport
+
+See [`lp_sensitivity`](@ref).
+"""
+struct SensitivityReport
+    rhs::Dict{ConstraintRef,Tuple{Float64,Float64}}
+    objective::Dict{VariableRef,Tuple{Float64,Float64}}
+end
+
+Base.getindex(s::SensitivityReport, c::ConstraintRef) = s.rhs[c]
+Base.getindex(s::SensitivityReport, x::VariableRef) = s.objective[x]
+
+"""
+    lp_sensitivity(model::Model; atol::Float64 = 1e-8)::SensitivityReport
+
+Given a linear program `model` containing a current optimal basis, return a
+[`SensitivityReport`](@ref) object, which maps:
+
+ - every constraint reference to a range over which the right-hand side of the
+    corresponding constraint can vary such that the basis remains optimal.
+    ```julia
+    model = lp_sensitivity(model)
+    dRHS_lo, dRHS_hi = model[c]
+    ```
+    Note: interval constraints are NOT supported.
+
+ - every variable reference to a range over which the objective coefficient of
+    the corresponding variable can vary such that the basis remains optimal.
+    ```julia
+    model = lp_sensitivity(model)
+    dx_lo, dx_hi = model[x]
+    ```
+
+`atol` is the primal/dual optimality tolerance, and should match the tolerance
+of the solver used to compute the basis.
+"""
+function lp_sensitivity(model::Model; atol::Float64 = 1e-8)
+    if !_is_lp(model)
+        error(
+            "Unable to compute LP sensitivity because model is not a linear " *
+            "program (or it contains interval constraints).",
+        )
+    elseif !has_values(model)
+        error("Unable to compute LP sensitivity: no primal solution available.")
+    elseif !has_duals(model)
+        error("Unable to compute LP sensitivity: no dual solution available.")
+    end
+
+    std_form = _standard_form_matrix(model)
+    basis = _standard_form_basis(model, std_form)
```

Comment:
Where will this function crash in the case that a basis is not available?

Reply:
It throws an error when querying `ConstraintBasisStatus`.